fero/process.py

```diff
@@ -0,0 +1,299 @@
+    modified = fields.DateTime(require=True)
+    latest_revision_version = fields.Integer(required=True)
+    username = fields.String(required=True)
+    process_type = fields.String(required=True, validate=FeroProcessTypes.validator())
+    product_type = fields.String(Required=False, allow_none=True)
+    kind = fields.String(required=True, validate=validate.OneOf(["process"]))
+    latest_ready_snapshot = fields.Nested(NestedSnapshotSchema, allow_none=True)
+    # TODO update this to a nested schema once the configuration settles more
+    data_config = fields.Dict(required=True)
+
+
+class Tag(FeroObject):
+    """High level object representing a process tag.
+
+    A tag is a single measurement type for a process and specifies metadata associated with this measurement since
+    as limits and cost.
+    """
+
+    schema_class = TagSchema
+
+    def __init__(self, process: "Process", client: "fero.Fero", data: dict):
+        self._process = process
+        super().__init__(client, data)
+
+    def __repr__(self):
+        return f"<Tag name={self.name} Process name={self._process}>"
+
+    __str__ = __repr__
+
+    def __eq__(self, other: object) -> bool:
+        return isinstance(other, Tag) and self._data == other._data
+
+    @classmethod
+    def tag_name(cls, instance: Union["Tag", str]) -> str:
+        return instance if isinstance(instance, str) else instance.name
+
+
+class Stage(FeroObject):
+    """High level object representing a process stage.
+
+    A tag is a discrete grouping of related tags that are part of process and describes metadata about how to process each tag
+    based on the type of process and configuration of the stage.
+    """
+
+    schema_class = StageSchema
+
+    def __init__(self, process: "Process", client: "fero.Fero", data: dict):
+        self._process = process
+        super().__init__(client, data)
+        self._data["tags"] = [
+            Tag(process, client, tdata) for tdata in self._data["tags"]
+        ]
+
+    def __repr__(self):
+        return f"<Stage name={self.name} Process name={self._process}>"
+
+    __str__ = __repr__
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, Stage) and self._data == other._data
+
+    @property
+    def tag_names(self) -> Sequence[str]:
+        """The names of the tags in the stage
+
+        :return: A list of tag names in the stage
+        :rtype: Sequence[str]
+        """
+        return [Tag.tag_name(t) for t in self.tags]
+
+
+class Process(FeroObject):
+    """High level object for interacting with a Process object defined in Fero.
+
+    A Process is a set of configurations that describe an industrial process and allow
+    Fero to combine, transform, and enrich raw data sources before the final data is
+    used in an Analysis.
+    """
+
+    schema_class = ProcessSchema
+
+    def __repr__(self):
+        return f"<Process name={self.name}>"
+
+    __str__ = __repr__
+
+    @property
+    @memoized(maxsize=1)
```

Comment:
Why only ` maxsize=1`?

Reply:
limits it to only cache 1 value (which should only be the original object).  Probably doesn't matter, but I didn't look to see if the implementation would reserve memory for more or anything....Also, stackoverflow just did it that way.